crates/primitives/src/crypto.rs

```diff
@@ -34,6 +34,22 @@ pub fn verify_schnorr_sig(sig: &Buf64, msg: &Buf32, pk: &Buf32) -> bool {
     sig.verify(&msg, &pk).is_ok()
 }
 
+#[cfg(target_os = "zkvm")]
+pub fn verify_schnorr_sig(sig: &Buf64, msg: &Buf32, pk: &Buf32) -> bool {
+    use k256::schnorr::{signature::hazmat::PrehashVerifier, Signature, VerifyingKey};
+    let sig = match Signature::try_from(sig.as_slice()) {
+        Ok(sig) => sig,
+        Err(_) => return false,
+    };
+
+    let vk = match VerifyingKey::from_bytes(pk.as_slice()) {
+        Ok(vk) => vk,
+        Err(_) => return false,
+    };
+
+    vk.verify_prehash(msg.as_slice(), &sig).is_ok()
+}
+
```

Comment:
Recommend adding documentation to both verification functions making it clear that the message _must_ have been properly hash processed.